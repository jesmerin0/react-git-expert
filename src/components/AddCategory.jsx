import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) =>{
        //console.log(target.value);
        setInputValue(target.value); 
    }

    const onSubmit = (event) =>{
        //para no propagar el evento y cargar el formulario
        event.preventDefault(); 

        const valueTrim = inputValue.trim();
        //limitar para que solo se guarde si tenemos un tama√±o de mas de 1 caracter
        if(valueTrim.length<=1) return;

        //console.log(inputValue);
        //setCategories(categories => [inputValue,...categories]);
        onNewCategory(valueTrim);
        setInputValue(''); 
    }

  return (
    <form onSubmit={onSubmit} aria-label="form">
        <input
            type="text"
            placeholder="Buscar gifs"
            value={inputValue}
            onChange={onInputChange}/>
    </form>        
  )
}


AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}