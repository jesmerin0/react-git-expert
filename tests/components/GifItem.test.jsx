import { render,screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => {
    
    const gifItem = {
        id:"VXJWhaO7afRe",
        title:"One Punch Man GIF",
        url:"https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=890ce5c6ejm1cvxz8ly7p4dh5npg8qq9ynhepl51qh4np5t6&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    }

    test('debe hacer match con el snapshot', () => { 
        const {container} = render(<GifItem {...gifItem}/>);
        expect(container).toMatchSnapshot();
     })

     test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        
        
        render(<GifItem {...gifItem}/>);
        //screen.debug();
        /*
        expect(screen.getByRole('img').src).toBe(gifItem.url);
        expect(screen.getByRole('img').alt).toBe(gifItem.title);
        */
       const { alt, src} = screen.getByRole('img');
       expect(src).toBe(gifItem.url);
       expect(alt).toBe(gifItem.title);
      })

      test('debe de mostrar el tÃ­tulo en el componente', () => { 
        
        
        render(<GifItem {...gifItem}/>);
        //screen.debug();
        expect(screen.getByText(gifItem.title)).toBeTruthy();
       
      })

 })