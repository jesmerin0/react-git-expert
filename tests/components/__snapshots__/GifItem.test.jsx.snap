// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Man GIF"
      src="https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=890ce5c6ejm1cvxz8ly7p4dh5npg8qq9ynhepl51qh4np5t6&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch Man GIF
    </p>
  </div>
</div>
`;
